### PR TITLE
misc shellcheck fixes

### PR DESCRIPTION
**Pull request by austin987**
This adds a .shellcheckrc (shellcheck config file for the project). I've currently set it to ignore using local in sh, since @williamh has told me that's policy (though I couldn't find a reference for that in the source, fyi).

It also fixes some misc issues. This doesn't fix everything, I need to test some others / fix more.

Note that if you want to use the .shellcheckrc, you'll need to build from git, since https://github.com/koalaman/shellcheck/commit/581bcc3907ab98e919a7dd60566810a928c46b95 isn't in a release yet.

### CODE REVIEW

**Comment by austin987** [comment]
Sorry for the confusion, wrong branch was initially pushed; but it's fixed now.

File: .shellcheckrc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Shellcheck configuration file
+
+# SC1090: Can't follow non-constant source. Use a directive to specify location
+# Ignore until https://github.com/koalaman/shellcheck/issues/769 is fixed upstream
+disable=SC1090
+
+# SC2039: In POSIX sh, local is undefined
```
Comment (vapier) [inline .shellcheckrc]: comments are full sentences, so put periods at the end of sentences

File: .shellcheckrc [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+# OpenRC allows local, which isn't POSIX
```
Comment (vapier) [inline .shellcheckrc]: link to https://github.com/koalaman/shellcheck/issues/1341

File: sh/binfmt.sh.in [hunk 3]
```diff
@@ -60,14 +60,14 @@ else
...
+		[ -d "$d" ] && for f in "${d}"/*.conf ; do
 			case "${f##*/}" in
 				systemd.conf|systemd-*.conf) continue;;
 			esac
-			[ -e $f ] && binfmt_basenames="${binfmt_basenames}\n${f##*/}"
+			[ -e "$f" ] && binfmt_basenames="${binfmt_basenames}\\n${f##*/}"
 		done # for f in ${d}
 	done # for d in ${binfmt_dirs}
-	binfmt_basenames="$(printf "${binfmt_basenames}\n" | sort -u )"
+	binfmt_basenames="$(printf "%s" "${binfmt_basenames}\\n" | sort -u )"
```
Comment (vapier) [inline sh/binfmt.sh.in]: you need `%b` here, not `%s`.  otherwise you're breaking this code.

File: test/units/sh_yesno [hunk 4]
```diff
@@ -27,7 +27,8 @@ for f in no NO No false FALSE False 0 ; do
 		echo "!$f!"
 	fi
 done
-: $(( ret += $tret ))
+eend $tret
```
Comment (vapier) [inline test/units/sh_yesno]: you didn't mean to add this line, so delete it

File: .shellcheckrc (hunk 1 above)
Comment (austin987) [reply .shellcheckrc]: Sure, fixed.

File: .shellcheckrc (hunk 2 above)
Comment (austin987) [reply .shellcheckrc]: Added a link, please let me know if you'd like to tweak the comment.

File: sh/binfmt.sh.in (hunk 3 above)
Comment (austin987) [reply sh/binfmt.sh.in]: Nice catch, thanks. Fixed.

File: test/units/sh_yesno (hunk 4 above)
Comment (austin987) [reply test/units/sh_yesno]: Yep, again, good catch, thanks.

**Pull request closed without merging**